### PR TITLE
issue_#209_number_of_games_double_in_PlayOff

### LINKED ISSUE #209: Number of games in Playoff doubled
1. Create Playoff tournament
2. Add several teams clicking Save after each team
3. Go to Schedule

Subtasks:

- [x]  #230 Modify adding/removing games to schedule 
- [x] #231 Change corresponded test or write new
- [x] #232 Manual QA test
**AR:** Each game is repeated twice

### PR DESCRIPTION
**Pull request by ViznyakS**
### Summary

In CreateSchedule modify method GetAllGamesInPlayOffTournament(to add scheduled game in new schedule), and change RemoveAllGamesInTournament( to remove all games from schedule, but not only games that have result)

closes #209
#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references
- [ ] When new property is added all related comparers are updated

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs [hunk 1]
```diff
@@ -317,7 +317,8 @@ public void SwapRounds(int tournamentId, byte firstRoundNumber, byte secondRound
         /// <param name="tournamentId">Identifier of the tournament</param>
         public void RemoveAllGamesInTournament(int tournamentId)
         {
-            var gamesToRemove = GetTournamentResults(tournamentId);
+           // var gamesToRemove = GetTournamentResults(tournamentId);
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs]: MAJOR Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sdiachen** [comment]
Checklist is for PR author in the first place

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs [hunk 2]
```diff
@@ -373,8 +374,11 @@ public void AddTeamsToTournament(ICollection<TeamTournamentAssignmentDto> groupT
             }
 
             var totalTeamCount = allTeams.Count + groupTeamCount;
+
+
             CreateSchedule(tournamentId, totalTeamCount);
 
+
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: what is it for?

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs [hunk 3]
```diff
@@ -737,31 +738,57 @@ private static int GetGamesCount(int teamsCount)
         private void CreateSchedule(int tournamentId, int allTeamsCount)
         {
             var tournament = Get(tournamentId);
+            var scheduledGames = _gameService.GetTournamentGames(tournamentId)
+                   .Where(tr => tr.Round == FIRST_ROUND);
+
             if (tournament.Scheme == TournamentSchemeEnum.PlayOff
-                && allTeamsCount > DONT_CREATE_SCHEDULE_TEAMS_COUNT)
+                && allTeamsCount > DONT_CREATE_SCHEDULE_TEAMS_COUNT
+                && scheduledGames.Count() != GetGamesCount(allTeamsCount))
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: What is th goal of this codition

Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: For example: 
when you add 3/4 team to tournament, schedule will be genarated for 4 team,
when you add 5/6/7/8 team to tournament, schedule will be genarated for 8 team ....
Goal is not to generate schedule, if the schedule is already generate for that count of team

I'm working on UTs)

Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: I see. I made mistake in condition.

Comment (ViznyakS) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: Is it enough to check the number of execute remove from _gameRepository?

Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: Didn't get last one.

My main concern is that you won't be able to tell me what this line checks in a month time.

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs [hunk 4]
```diff
@@ -737,31 +738,57 @@ private static int GetGamesCount(int teamsCount)
         private void CreateSchedule(int tournamentId, int allTeamsCount)
         {
             var tournament = Get(tournamentId);
+            var scheduledGames = _gameService.GetTournamentGames(tournamentId)
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: Do you need to call that for non-playoff tournaments?

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs [hunk 5]
```diff
@@ -36,6 +36,7 @@ public class TournamentService : ITournamentService
         private const int MIN_TEAMS_TO_PLAY_ONE_ROUND = 2;
         private const int GAMES_TO_PLAY_ONE_ROUND = 2;
         private const int DONT_CREATE_SCHEDULE_TEAMS_COUNT = 1;
+        private const int FIRST_ROUND = 1;
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: MAJOR Remove the unused private field 'FIRST_ROUND'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1144)

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs (hunk 5 above)
Comment (sdiachen) [reply src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: I ❤️ Sonar))))

File: src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs [hunk 6]
```diff
@@ -737,31 +737,34 @@ private static int GetGamesCount(int teamsCount)
...
+                && allTeamsCount > DONT_CREATE_SCHEDULE_TEAMS_COUNT)             
             {
                 var gamesToAdd = GetAllGamesInPlayOffTournament(tournamentId, allTeamsCount);
                 _gameService.RemoveAllGamesInTournament(tournamentId);
                 _gameService.AddGames(gamesToAdd);
             }
         }
 
-        private static List<Game> GetAllGamesInPlayOffTournament(int tournamentId, int teamsCount)
+        private List<Game> GetAllGamesInPlayOffTournament(int tournamentId, int teamsCount)
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs]: should it be static?

**Pull request merged**